File: Semana 4/Dia 4/Libreria/models/categoria.py
from config.base_datos import bd


class CategoriaModel(bd.Model):
    __tablename__ = "t_categoria"
    categoriaId = bd.Column(name="categoria_id", type_=bd.Integer,
                            primary_key=True, autoincrement=True, nullable=False, unique=True)
    categoriaDescripcion = bd.Column(
        name="categoria_descripcion", type_=bd.String(45), unique=True, nullable=False)

File: Semana 7/CRUD_SEQUELIZE_EXPRESS/src/models/Categoria.js
const { DataTypes } = require("sequelize");
const { conexion } = require("../config/Sequelize");
// https://sequelize.org/master/manual/model-basics.html#column-options
module.exports = categoria_model = () => {
  return (categoria = conexion.define(
    // en el caso de una relacion hijo - padre en el nombre del modelo si la ultima letra es la A la retirará y la remplazara por "um", si es "S" la quitara y la dejara en singular y las demas letras las respetara
    "categorias",
    {
      categoriaId: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        allowNull: false, // no puede permitir valores nulos
        autoIncrement: true, // es autoincrementable
        field: "cat_id", // el nombre de esa columna en la tabla de la bd
      },
      categoriaNombre: {
        type: DataTypes.STRING(45),
        unique: true,
        allowNull: false,
        field: "cat_nombre",
      },
    },
    {
      tableName: "t_categoria",
      timestamps: false,
    }
  ));
};

// module.exports = categoria_model
